src/objects/Gate.gd

```diff
@@ -39,8 +39,10 @@ func open() -> void:
 		return
 	$AnimatedSprite.play("open")
 	tile_map.set_cellv(grid_coords, tile_map.tile_set.find_tile_by_name("Base"))
-	z_index = 1
 	is_open = true
+	while $AnimatedSprite.frame <= 4:
+		yield($AnimatedSprite, "frame_changed")
```

Comment:
What does yield do in this case? Just curious

Reply:
Makes it so that it becomes "flat" at the moment the sprite is flush with the ground in the animation (rather than right as it's opened)

Reply:
I mean like what the yield statement does specifically, too lazy to google lmao

Reply:
forgor💀 to respond - basically, it stops execution of this function and waits until that signal is emitted to resume it. So here, it just keeps waiting for every time the frame changes to check for the condition and continue